File: documentation/user/en/solve/examples/filtering-products-in-category/product-listing-with-facets-and-sorting.cs
EvitaResponse<ISealedEntity> entities = evita.QueryCatalog(
	"evita",
	session => session.QuerySealedEntity(
        Query(
        	Collection("Product"),
        	FilterBy(
        		EntityLocaleEquals(CultureInfo.GetCultureInfo("en")),
        		HierarchyWithin(
        			"categories",
        			AttributeEquals("url", "/en/smartwatches")
        		),
        		AttributeEquals("status", "ACTIVE"),
        		Or(
        			AttributeInRangeNow("validity"),
        			AttributeIs("validity", Null)
        		),
        		ReferenceHaving(
        			"stocks",
        			AttributeGreaterThan("quantityOnStock", 0)
        		),
        		PriceInCurrency(new Currency("EUR")),
        		PriceInPriceLists("basic"),
        		PriceValidInNow(),
        		UserFilter(
        			FacetHaving(
        				"brand",
        				EntityPrimaryKeyInSet(66465)
        			),
        			PriceBetween(50m, 400m)
        		)
        	),
        	OrderBy(
        		AttributeNatural("order", Asc)
        	),
        	Require(
        		EntityFetch(
        			AttributeContent("name"),
        			ReferenceContentWithAttributes(
        				"stocks",
        				AttributeContent("quantityOnStock")
        			),
        			PriceContentRespectingFilter("reference")
        		),
        		FacetSummaryOfReference(
        			"brand",
        			Impact,
        			OrderBy(
        				AttributeNatural("name", Asc)
        			),
        			EntityFetch(
        				AttributeContent("name")
        			)
        		),
        		FacetSummaryOfReference(
        			"parameterValues",
        			Impact,
        			FilterGroupBy(
        				AttributeEquals("isVisibleInFilter", true)
        			),
        			OrderBy(
        				AttributeNatural("order", Asc)
        			),
        			OrderGroupBy(
        				AttributeNatural("order", Asc)
        			),
        			EntityFetch(
        				AttributeContent("name")
        			),
        			EntityGroupFetch(
        				AttributeContent("name")
        			)
        		),
        		PriceHistogram(10, Standard),
        		Page(1, 16)
        	)
        )
	)
);

File: documentation/user/en/solve/examples/filtering-products-in-category/product-listing-with-facets-and-sorting.java
final EvitaResponse<SealedEntity> entities = evita.queryCatalog(
	"evita",
	session -> {
		return session.querySealedEntity(
			query(
				collection("Product"),
				filterBy(
					entityLocaleEquals(Locale.forLanguageTag("en")),
					hierarchyWithin(
						"categories",
						attributeEquals("url", "/en/smartwatches")
					),
					attributeEquals("status", "ACTIVE"),
					or(
						attributeInRangeNow("validity"),
						attributeIs("validity", NULL)
					),
					referenceHaving(
						"stocks",
						attributeGreaterThan("quantityOnStock", 0)
					),
					priceInCurrency(Currency.getInstance("EUR")),
					priceInPriceLists("basic"),
					priceValidInNow(),
					userFilter(
						facetHaving(
							"brand",
							entityPrimaryKeyInSet(66465)
						),
						priceBetween(new BigDecimal("50"), new BigDecimal("400"))
					)
				),
				orderBy(
					attributeNatural("order", ASC)
				),
				require(
					entityFetch(
						attributeContent("name"),
						referenceContentWithAttributes(
							"stocks",
							attributeContent("quantityOnStock")
						),
						priceContentRespectingFilter("reference")
					),
					facetSummaryOfReference(
						"brand",
						IMPACT,
						orderBy(
							attributeNatural("name", ASC)
						),
						entityFetch(
							attributeContent("name")
						)
					),
					facetSummaryOfReference(
						"parameterValues",
						IMPACT,
						filterGroupBy(
							attributeEquals("isVisibleInFilter", true)
						),
						orderBy(
							attributeNatural("order", ASC)
						),
						orderGroupBy(
							attributeNatural("order", ASC)
						),
						entityFetch(
							attributeContent("name")
						),
						entityGroupFetch(
							attributeContent("name")
						)
					),
					priceHistogram(10, STANDARD),
					page(1, 16)
				)
			)
		);
	}
);
